filters/bevel/src/bevel.frag

```diff
@@ -0,0 +1,23 @@
+precision mediump float;
+
+varying vec2 vTextureCoord;
+uniform sampler2D uSampler;
+uniform vec4 filterArea;
+
+uniform float transformX;
+uniform float transformY;
+uniform vec3 lightColor;
+uniform float lightAlpha;
+uniform vec3 shadowColor;
+uniform float shadowAlpha;
+
+void main(void) {
+    vec2 transform = vec2(1.0 / filterArea) * vec2(transformX, transformY);
+    vec4 color = texture2D(uSampler, vTextureCoord);
+    float light = texture2D(uSampler, vTextureCoord - transform).a;
+    float shadow = texture2D(uSampler, vTextureCoord + transform).a;
+
+    color.rgb = mix(color.rgb, lightColor, clamp(color.a - light, 0.0, 1.0) * lightAlpha);
+    color.rgb = mix(color.rgb, shadowColor, clamp(color.a - shadow, 0.0, 1.0) * shadowAlpha);
```

Comment:
color is premultiplied. Those formulas are kinda wrong for alpha that is not 0.0 or 1.0